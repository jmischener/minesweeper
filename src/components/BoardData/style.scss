.message0 .message1 {
    display: none;
}

.message2 {
    font-size: 40px;
    width: 100%;
    height: 65px;
    color: red;
    margin-top: 15px;
}

.message3 {
    font-size: 40px;
    width: 100%;
    height: 65px;
    color: green;
    margin-top: 15px;
}