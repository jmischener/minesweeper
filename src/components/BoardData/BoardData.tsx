import React from "react";
import { eBoardState, IBoardData, RootState } from "../../types/types";
import { useSelector } from "react-redux";
import Alert from "@material-ui/lab/Alert";

import "./style.scss";

interface BoardDataProps {
  size: number;
}
export const BoardData: React.FunctionComponent<BoardDataProps> = (props) => {
  const data = useSelector<RootState, IBoardData>((state) => state.game.data);

  const renderMessage = () => {
    switch (data.step) {
      case eBoardState.Lose:
        return <Alert severity="error">Perdiste!</Alert>;
      case eBoardState.Win:
        return <Alert severity="success">Ganaste!</Alert>;
      default:
        return "";
    }
  };

  return (
    <div style={{ width: props.size * 30 + 2 }}>
      <div>{renderMessage()}</div>
      <div>
        {data.step !== eBoardState.Clean && (
          <Alert severity="info">
            {`Cantidad Bombas:${data.cantBombs - data.cantFlags}`}
          </Alert>
        )}
      </div>
    </div>
  );
};
