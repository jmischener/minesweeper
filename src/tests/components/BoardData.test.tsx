import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { BoardData } from "../../components/BoardData/BoardData";

const mockStore = configureStore();

const initState = {
  game: {
    board: {
      squares: [
        { isBomb: false, state: 0, value: 1 },
        { isBomb: false, state: 0, value: 1 },
        { isBomb: true, state: 0, value: 0 },
        { isBomb: false, state: 0, value: 1 },
      ],
      size: 2,
    },
    data: { step: 1, cantBombs: 1, cantFlags: 0, cantShow: 3 },
  },
};

const loseState = {
  game: {
    board: {
      squares: [
        { isBomb: false, state: 0, value: 1 },
        { isBomb: false, state: 0, value: 1 },
        { isBomb: true, state: 1, value: 0 },
        { isBomb: false, state: 0, value: 1 },
      ],
      size: 2,
    },
    data: { step: 2, cantBombs: 1, cantFlags: 0, cantShow: 3 },
  },
};

const winState = {
  game: {
    board: {
      squares: [
        { isBomb: false, state: 1, value: 1 },
        { isBomb: false, state: 1, value: 1 },
        { isBomb: true, state: 0, value: 0 },
        { isBomb: false, state: 1, value: 1 },
      ],
      size: 2,
    },
    data: { step: 3, cantBombs: 1, cantFlags: 0, cantShow: 0 },
  },
};

describe("Testeo de Componente MineSweeper", () => {
  test("Debe mostrarse sin mensajes y con la informaciÃ³n correcta", () => {
    let store = mockStore(() => initState);
    store.dispatch = jest.fn();

    const wrapper = mount(
      <Provider store={store}>
        <BoardData size={2} />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot("BoardData-InGame");
  });

  test("Debe mostrarse mensaje de ganaste", () => {
    let store = mockStore(() => winState);
    store.dispatch = jest.fn();

    const wrapper = mount(
      <Provider store={store}>
        <BoardData size={2} />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot("BoardData-Win");
  });

  test("Debe mostrarse mensaje de perdiste", () => {
    let store = mockStore(() => loseState);
    store.dispatch = jest.fn();

    const wrapper = mount(
      <Provider store={store}>
        <BoardData size={2} />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot("BoardData-Lose");
  });
});

export {};
