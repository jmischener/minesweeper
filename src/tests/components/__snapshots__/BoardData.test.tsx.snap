// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeo de Componente MineSweeper Debe mostrarse mensaje de ganaste: BoardData-Win 1`] = `
<div
  style={
    Object {
      "width": 62,
    }
  }
>
  <div>
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess MuiPaper-elevation0"
      role="alert"
    >
      <div
        className="MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message"
      >
        Ganaste!
      </div>
    </div>
  </div>
  <div>
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
      role="alert"
    >
      <div
        className="MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message"
      >
        Cantidad Bombas:1
      </div>
    </div>
  </div>
</div>
`;

exports[`Testeo de Componente MineSweeper Debe mostrarse mensaje de perdiste: BoardData-Lose 1`] = `
<div
  style={
    Object {
      "width": 62,
    }
  }
>
  <div>
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
      role="alert"
    >
      <div
        className="MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message"
      >
        Perdiste!
      </div>
    </div>
  </div>
  <div>
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
      role="alert"
    >
      <div
        className="MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message"
      >
        Cantidad Bombas:1
      </div>
    </div>
  </div>
</div>
`;

exports[`Testeo de Componente MineSweeper Debe mostrarse sin mensajes y con la informaci√≥n correcta: BoardData-InGame 1`] = `
<div
  style={
    Object {
      "width": 62,
    }
  }
>
  <div />
  <div>
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
      role="alert"
    >
      <div
        className="MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message"
      >
        Cantidad Bombas:1
      </div>
    </div>
  </div>
</div>
`;
